### PR TITLE
Replaces astropy-helper with sphinx-astropy 

### PR DESCRIPTION
**Pull request by yashkgp**
Solves Issue #2489

### CODE REVIEW

File: .rtd-environment.yml [hunk 1]
```diff
@@ -61,4 +61,4 @@ dependencies:
 - glymur
 - pip:
     - sunpy-sphinx-theme
-    - git+https://github.com/sphinx-gallery/sphinx-gallery
```
Comment (Cadair) [inline .rtd-environment.yml]: sphinx-gallery and sphinx-astropy do two different things we are going to need both :smile:

File: docs/conf.py [hunk 2]
```diff
@@ -29,14 +29,7 @@
 import datetime
 import sys
 
-ON_RTD = os.environ.get('READTHEDOCS') == 'True'
```
Comment (Cadair) [inline docs/conf.py]: I think we are still going to need all of this RTD stuff.

File: docs/conf.py [hunk 3]
```diff
@@ -29,14 +29,7 @@
...
-ON_TRAVIS = os.environ.get('TRAVIS') == 'true'
-
-if ON_RTD:
-    os.environ['SUNPY_CONFIGDIR'] = '/home/docs/'
-    os.environ['HOME'] = '/home/docs/'
-    os.environ['LANG'] = 'C'
-    os.environ['LC_ALL'] = 'C'
+import astropy
```
Comment (Cadair) [inline docs/conf.py]: I don't think we need this.

**Comment by nabobalis** [comment]
I think down the road we can remove sphinx-gallery since its one of the dependencies for sphinx-astropy. But since it has't had a release for a while, we still need to use master.

**Comment by Cadair** [comment]
> /home/travis/build/sunpy/sunpy/build/lib.linux-x86_64-3.6/sunpy/spectra/spectrum.py:docstring of sunpy.spectra.spectrum.Spectrum:34: WARNING: Inline strong start-string without end-string.

File: docs/conf.py [hunk 4]
```diff
@@ -174,14 +166,13 @@
 napoleon_google_docstring = False
 
 # -- Options for the edit_on_github extension ----------------------------------------
-extensions.remove('astropy_helpers.extern.numpydoc')
```
Comment (Cadair) [inline docs/conf.py]: we still need to remove this, jut from `astropy_sphinx` instead. It's possible this is what's causing the doc warning.

**Comment by Cadair** [comment]
I have just had a look through the `conf.py` file, and while this is very much *not* your fault, it's a bit of a mess, and could do with a tidy. This seems like an opportune time to tidy it up where appropriate.

Firstly, a quick bit of background. Sphinx imports `conf.py` and then uses any variables that are defined in that file to configure itself, i.e. it looks for a variable named `extensions` in `conf.py` to know what extensions to load (it will not use any variables in the file it is not expecting to be there).

How we organise our `conf.py` file is that we import the one from `astropy_helpers` or now `sphinx_astropy` and then modify things as we need to. This means we re-define some variables outright, and append or delete things from the lists and dictionaries we have imported from the astropy config.

What would be nice to do is to look through the [`conf/v1.py`](https://github.com/astropy/sphinx-astropy/blob/master/sphinx_astropy/conf/v1.py) file in `sphinx_astropy` and find places where we duplicate things in that file or re-define things to be the same etc. Obvious ones I can see are `intersphinx_mapping` and `on_rtd`.

Then also feel free to re-format our `conf.py` to be as succinct as possible. The main changes we make to the default config in `astropy_sphinx` are:

* Use our HTML theme
* remove h5py from intersphinx and add sqlalchemy, pandas and skimage.
* use napoleon instead of numpy doc
* extra config for sphinx-gallery.

Also worth looking at is the [`conf.py`](https://github.com/sunpy/sunpy-sphinx-theme/blob/master/sunpy_sphinx_theme/conf.py) for the sunpy theme, as it looks like we are re-defining one or two variables from that as well.

Thanks!

**Comment by yashkgp** [comment]
Thanks for the help @Cadair ,will update the `conf.py` ASAP.

File: docs/conf.py [hunk 5]
```diff
@@ -174,14 +166,13 @@
...
 extensions.append('sphinx.ext.napoleon')
```
Comment (yashkgp) [inline docs/conf.py]: Hey @Cadair ,Why `sphinx.ext.napoleon` is appended 2 times ,does this serve any purpose ? Also `numpydocs` should be removed from `extensions` right ?

File: docs/conf.py (hunk 4 above)
Comment (yashkgp) [reply docs/conf.py]: But extension in `astropy_sphinx` does not contain `astropy_helpers.extern.numpydoc`, will raise an error if removed . Also `numpydocs` should be removed right ?

Comment (yashkgp) [reply docs/conf.py]: @Cadair ping

Comment (Cadair) [reply docs/conf.py]: yeah in `astropy_sphinx` it looks like it's just `numpydoc` we need to pop (https://github.com/astropy/sphinx-astropy/blob/master/sphinx_astropy/conf/v1.py#L135)

File: docs/conf.py [hunk 6]
```diff
@@ -174,6 +172,9 @@
 napoleon_google_docstring = False
 extensions += ['sphinx_astropy.ext.edit_on_github', 'sphinx.ext.doctest', 'sphinx.ext.githubpages']
 
+# Remove numpydoc
```
Comment (Cadair) [inline docs/conf.py]: Can you put this directly above the ` extensions.append('sphinx.ext.napoleon')` line?

Also the napoleon config seems to be duplicated?  Can you remove the one not under the "swap to napoleon" comment?

**Comment by yashkgp** [comment]
Done

File: CHANGELOG.rst [hunk 7]
```diff
@@ -42,6 +42,8 @@ Bug Fixes
 - Updates MapCube to access the correct properties of the namedtuple SpatialPair [#2297]
 - Fixed TimeSeries test failures due to missing test files [#2273]
 - Refactored a GOES test to avoid a Py3.6 issue [#2276]
+- The documentation build now uses the Sphinx configuration from sphinx-astropy 
+  rather than from astropy-helpers.
```
Comment (Cadair) [inline CHANGELOG.rst]: this needs the pr number in square brackets at the end.

File: docs/conf.py [hunk 8]
```diff
@@ -166,22 +156,16 @@
...
+extensions.remove('numpydoc')
 extensions.append('sphinx.ext.napoleon')
 
 # Disable having a separate return type row
 napoleon_use_rtype = False
 # Disable google style docstrings
 napoleon_google_docstring = False
 
-# -- Options for the edit_on_github extension ----------------------------------------
```
Comment (Cadair) [inline docs/conf.py]: This comment should still be here.

File: requirements/docs.txt [hunk 9]
```diff
@@ -7,3 +7,4 @@
 sphinx
 sunpy-sphinx-theme
 sphinx-gallery
+sphinx_astropy
```
Comment (Cadair) [inline requirements/docs.txt]: the pip name is `sphinx-astropy` not `sphinx_astropy`.

(yes, yes it's a very confusing convention lol)

File: docs/conf.py (hunk 8 above)
Comment (yashkgp) [reply docs/conf.py]: Done

File: docs/conf.py (hunk 2 above)
Comment (nabobalis) [reply docs/conf.py]: Should we check if we do need this?

File: docs/conf.py [hunk 10]
```diff
@@ -29,10 +29,15 @@
...
+try:
+    from sphinx_astropy.conf.v1 import *
+except ImportError:
+    print('ERROR: the documentation requires the sphinx-astropy package to be installed')
+    sys.exit(1)
+
 ON_TRAVIS = os.environ.get('TRAVIS') == 'true'
 
-if ON_RTD:
+if on_rtd:
```
Comment (nabobalis) [inline docs/conf.py]: Should be `ON_RTD`

File: docs/conf.py (hunk 8 above)
Comment (nabobalis) [reply docs/conf.py]: This was moved to the wrong part of the file?

File: docs/conf.py [hunk 11]
```diff
@@ -200,7 +185,7 @@
 
 # -- Options for the Sphinx gallery -------------------------------------------
 
-if ON_RTD and os.environ.get('READTHEDOCS_PROJECT').lower() != 'sunpy':
+if on_rtd and os.environ.get('READTHEDOCS_PROJECT').lower() != 'sunpy':
```
Comment (nabobalis) [inline docs/conf.py]: `ON_RTD`

File: sunpy/database/tests/test_database.py [hunk 12]
```diff
@@ -902,7 +902,8 @@ def test_fetch(database, download_query, tmpdir):
     database.undo()
     assert len(database) == 0
     database.redo()
-    assert len(database) == 4
+    # Make this resilitent to vso changes while we chase this up with VSO 2018-03-07
+    assert len(database) in (2, 4)
```
Comment (nabobalis) [inline sunpy/database/tests/test_database.py]: THIS IS MAGIC, WHY DIDNT I DO THIS.

File: docs/conf.py (hunk 10 above)
Comment (Cadair) [reply docs/conf.py]: No, we now import this from sphinx_astropy and it's lower case.

File: docs/conf.py (hunk 2 above)
Comment (Cadair) [reply docs/conf.py]: See comment below, we are now importing it from sphinx astropy

File: docs/conf.py (hunk 10 above)
Comment (nabobalis) [reply docs/conf.py]: OH FAIR ENOUGH I SHOULD HAVE CHECKED

**Comment by Cadair** [comment]
Thanks @yashkgp this is awesome!

**Pull request merged**